svchost/disco/disco.go

```diff
@@ -39,6 +39,10 @@ var httpTransport = cleanhttp.DefaultPooledTransport() // overridden during test
 type Disco struct {
 	hostCache map[svchost.Hostname]Host
 	credsSrc  auth.CredentialsSource
+
+	// Transport is a custom http.Transport to use.
+	// A package default is used if this is nil.
+	Transport *http.Transport
```

Comment:
Seems fair enough... maybe in this case we can drop the global variable here and just make the tests _within_ this package use this mechanism too, for consistency. Can defer that to a separate change though, since it's not too important.

Reply:
we should still use a package-level transport, so that transport connections can be reused. There's a few of these getting scattered around plugin and module packages. Maybe terraform needs a package that returns single global transport with a known state 🤔